system/include/compat/sys/random.h

```diff
@@ -0,0 +1,9 @@
+#ifdef __cplusplus
```

Comment:
Maybe add a comment about why we are adding this.. how it helps?

Reply:
I don't understand. The `__cplusplus` pattern? Lots of headers have this, like these siblings in the same dir:
```
system/include/compat/malloc.h
system/include/compat/netdb.h
system/include/compat/stdarg.h
system/include/compat/stdlib.h
system/include/compat/string.h
system/include/compat/time.h
system/include/compat/xlocale.h
```

Reply:
Sorry no I meant a comment about why this header is added by emscripten when its not part of musl?  What is the purpose of adding getransom?   I guess that answer is something like "to get ransom bytes without depending on the filesytem stuff which is good for code size and complexity in the generated output" .. or something along those lines :)

Reply:
Ok, adding now.

The bigger question is maybe why isn't this in musl, given it seems to be a Linux unistd.h thing..? Looks like musl has added it meanwhile actually, https://github.com/ifduyue/musl/commit/82f176803ae07e34229906d5c7c62889e665dc97#diff-2fc78454dd4e341aace02870b1e0c8c9 - but using a syscall, which would adds some unnecessary redirection for us, ando it's not where libc++ looks for it (in unistd as opposed to sys/random).